src/System.Management.Automation/engine/CommandSearcher.cs

```diff
@@ -459,49 +459,61 @@ private CommandInfo GetNextFromPath()
-                        _commandName,
-                        providerException.Message);
+                    try
+                    {
+                        Provider.CmdletProvider providerInstance;
+                        resolvedPaths =
+                            _context.LocationGlobber.GetGlobbedProviderPathsFromMonadPath(_commandName, false, out provider, out providerInstance);
+                    }
+                    catch (ItemNotFoundException)
+                    {
+                        CommandDiscovery.discoveryTracer.TraceError(
+                            "The path could not be found: {0}",
+                            _commandName);
+                    }
+                    catch (DriveNotFoundException)
+                    {
+                        CommandDiscovery.discoveryTracer.TraceError(
+                            "A drive could not be found for the path: {0}",
+                            _commandName);
+                    }
+                    catch (ProviderNotFoundException)
+                    {
+                        CommandDiscovery.discoveryTracer.TraceError(
+                            "A provider could not be found for the path: {0}",
+                            _commandName);
+                    }
+                    catch (InvalidOperationException)
+                    {
+                        CommandDiscovery.discoveryTracer.TraceError(
+                            "The path specified a home directory, but the provider home directory was not set. {0}",
+                            _commandName);
+                    }
+                    catch (ProviderInvocationException providerException)
+                    {
+                        CommandDiscovery.discoveryTracer.TraceError(
+                            "The provider associated with the path '{0}' encountered an error: {1}",
+                            _commandName,
+                            providerException.Message);
+                    }
+                    catch (PSNotSupportedException)
+                    {
+                        CommandDiscovery.discoveryTracer.TraceError(
+                            "The provider associated with the path '{0}' does not implement ContainerCmdletProvider",
+                            _commandName);
+                    }
                 }
-                catch (PSNotSupportedException)
+
+                // Try literal path resolution if wildcards are disable or wildcard search failed
+                if (resolvedPaths.Count == 0)
                 {
-                    CommandDiscovery.discoveryTracer.TraceError(
-                        "The provider associated with the path '{0}' does not implement ContainerCmdletProvider",
-                        _commandName);
+                    string path = _context.LocationGlobber.GetProviderPath(_commandName, out provider);
+                    resolvedPaths = new System.Collections.ObjectModel.Collection<string>();
```

Comment:
You no longer need to create a new `resolvedPaths` object, and can just add to the existing collection.